zanio/PropertyPro-lite
# #166899578 build admin database and admin validation

[ Finish #166899578]

***What does this PR do?***
It builds the database and middleware for admin users

**Description of Task to be completed?**
none

**How should this be manually tested?**
clone this branch and run ```npm start``` and check postman at the url ```http://localhost:3300/api/v1/auth/register``` with the required input field from the pivotal stories
Any background context you want to provide?
None

**What are the relevant pivotal tracker stories?**
As admin  
I should already exist in the database
So I can be given the admin badge

### Acceptance Criteria

-   Given admin
-   When admin is about to register  
    `POST /api/v1/auth/register`
-   Then middleware function should find the admin name and email from the database to see  if it exist and return the object

Request:
```json
{
    "email" : String,
    "firstName" : String,
    "lastName" : String,
    "gender" : String,
    "password" : String,
    "phoneNumber" : String,
    "address" : String,
    
}
```

Response:

```json
{
   { 
    "first_name": "Aniefiok",
    "last_name": "Akpan",
    "is_Admin": true,
    "email": "akp.ani@yahoo.com"
    }
}
```

Screenshots (if appropriate)
![Screenshot (25)](https://user-images.githubusercontent.com/49557553/60168878-88411f00-97fd-11e9-9c95-54cd5bb1c19d.png)
![Screenshot (24)](https://user-images.githubusercontent.com/49557553/60168880-88411f00-97fd-11e9-8918-a398d17457a1.png)
